From 1684c855b69da923e44acf3776c4a3704dc1de32 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Mon, 23 Jul 2018 10:58:04 +0800
Subject: [PATCH 4218/5242] MLK-19001-2 ARM64: dts: imx8mq: fill
 init-on-arrary for clk

commit  f14e02ab82dfb6493c620b4c68a643657a4bb203 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fill init-on-array property to let root cell Linux enable
these clks, because we need differnet clocks being init on compared with
clks_init_on in clk-imx8mq.c.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
(cherry picked from commit 8ea027403874acaa15953fee1531961335c2a6e8)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mq-evk-root.dts     |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-root.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-root.dts
index cff47f6..13ce6b7 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-root.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-root.dts
@@ -25,6 +25,17 @@
 	arm,psci-suspend-param = <0x0>;
 };
 
+&clk {
+	init-on-array = <IMX8MQ_CLK_DRAM_CORE IMX8MQ_CLK_AHB_CG
+			 IMX8MQ_CLK_NOC_CG IMX8MQ_CLK_NOC_APB_CG
+			 IMX8MQ_CLK_USB_BUS_CG
+			 IMX8MQ_CLK_MAIN_AXI_CG IMX8MQ_CLK_A53_CG
+			 IMX8MQ_CLK_AUDIO_AHB_DIV IMX8MQ_CLK_TMU_ROOT
+			 IMX8MQ_CLK_DRAM_APB_DIV
+			 IMX8MQ_CLK_NOC_CG IMX8MQ_CLK_NOC_APB_CG
+			 IMX8MQ_CLK_NAND_USDHC_BUS_CG>;
+};
+
 &iomuxc {
 	imx8mq-evk {
 		/*
-- 
1.7.9.5

